ui/src/treeDataProvider/AzureTreeDataProvider.ts

```diff
@@ -171,15 +171,24 @@ export class AzureTreeDataProvider implements TreeDataProvider<IAzureNode>, Disp
     }
 
     public async findNode(id: string): Promise<IAzureNode | undefined> {
-        let children: IAzureNode[] = await this.getChildren();
+        let nodes: IAzureNode[] = await this.getChildren();
 
         // tslint:disable-next-line:no-constant-condition
         while (true) {
-            const node: IAzureNode | undefined = children.find((child: IAzureNode) => id.startsWith(child.id));
-            if (node && node.id === id) {
-                return node;
-            } else if (node instanceof AzureParentNode) {
-                children = await node.getCachedChildren();
+            let ancestorNode: AzureParentNode | undefined;
+            for (const node of nodes) {
+                if (node.id === id) {
+                    return node;
+                } else if (id.startsWith(`${node.id}/`) && node instanceof AzureParentNode) {
+                    // Append '/' to 'node.id' when checking 'startsWith' to ensure its actually an ancenstor, rather than a node at the same level that _happens_ to start with the same id
+                    // For example, two databases named 'test' and 'test1' as described in this issue: https://github.com/Microsoft/vscode-cosmosdb/issues/488
+                    ancestorNode = node;
```

Comment:
nit: seems you could get rid of ancestorNode and just do nodes = await... right after the check against starts With.

Reply:
I still need some way to know when I _didn't_ find an ancestor, but you right I can call `nodes = await ...` here. Refactored a bit